web-app/js/portal/details/DataCollectionDetailsPanel.js

```diff
@@ -14,18 +14,7 @@ Portal.details.DataCollectionDetailsPanel = Ext.extend(Ext.Panel, {
         var tabPanelForLayer = this._initSubsetItemsTabPanel(cfg);
 
         this.createTools(cfg.dataCollection.getLayerState());
-
-        cfg.dataCollection.getLayerState().on('loadstart', function() {
-            this._onLayerLoadStart();
-        }, this);
-
-        cfg.dataCollection.getLayerState().on('loadend', function() {
-            this._onLayerLoadEnd();
-        }, this);
-
-        cfg.dataCollection.getLayerState().on('tileerror', function() {
-            this._onLayerLoadError();
-        }, this);
+        this.addListeners(cfg.dataCollection.getLayerState());
```

Comment:
It's nice how you've extracted this method out (and the new `removeListeners()` method). It makes the constructor much tidier.